pants-plugins/uses_services/mongo_rules.py

```diff
@@ -0,0 +1,283 @@
+    level=LogLevel.DEBUG,
+)
+async def mongo_is_running_for_pytest(
+    request: PytestUsesMongoRequest,
+) -> PytestPluginSetup:
+    # TODO: delete these comments once the Makefile becomes irrelevant.
+    #       the comments explore how the Makefile prepares to run and runs tests
+
+    # The st2-test database gets dropped between (in Makefile based testing):
+    #   - each component (st2*/ && various config/ dirs) in Makefile
+    #   - DbTestCase/CleanDbTestCase setUpClass
+
+    # Makefile
+    #    .run-unit-tests-coverage (<- .combine-unit-tests-coverage <- .coverage.unit <- .unit-tests-coverage-html <- ci-unit <- ci)
+    #        echo "----- Dropping st2-test db -----"
+    #        mongo st2-test --eval "db.dropDatabase();"
+    #        for component in $(COMPONENTS_TEST)
+    #            nosetests $(NOSE_OPTS) -s -v $(NOSE_COVERAGE_FLAGS) $(NOSE_COVERAGE_PACKAGES) $$component/tests/unit
+
+    # this will raise an error if mongo is not running
+    _ = await Get(MongoIsRunning, UsesMongoRequest())
+
+    return PytestPluginSetup()
+
+
+@rule(
+    desc="Test to see if mongodb is running and accessible.",
+    level=LogLevel.DEBUG,
+)
+async def mongo_is_running(
+    request: UsesMongoRequest, platform: Platform
+) -> MongoIsRunning:
+    script_path = "./is_mongo_running.py"
+
+    # pants is already watching this directory as it is under a source root.
+    # So, we don't need to double watch with PathGlobs, just open it.
+    with open(is_mongo_running_full_path, "rb") as script_file:
+        script_contents = script_file.read()
+
+    script_digest, mongoengine_pex = await MultiGet(
+        Get(Digest, CreateDigest([FileContent(script_path, script_contents)])),
+        Get(
+            VenvPex,
+            PexRequest(
+                output_filename="mongoengine.pex",
+                internal_only=True,
+                requirements=PexRequirements({"mongoengine", "pymongo"}),
+            ),
+        ),
+    )
+
+    result = await Get(
+        FallibleProcessResult,
+        VenvPexProcess(
+            mongoengine_pex,
+            argv=(
+                script_path,
+                request.db_host,
+                str(request.db_port),
+                request.db_name,
+                str(request.db_connection_timeout),
+            ),
+            input_digest=script_digest,
+            description="Checking to see if Mongo is up and accessible.",
+            # this can change from run to run, so don't cache results.
+            cache_scope=ProcessCacheScope.PER_SESSION,
+            level=LogLevel.DEBUG,
+        ),
+    )
+    is_running = result.exit_code == 0
+
+    if is_running:
+        return MongoIsRunning()
+
+    # mongo is not running, so raise an error with instructions.
+
+    if platform.distro in ["centos", "rhel"] or "rhel" in platform.distro_like:
+        instructions = dedent(
+            """\
+            If mongo is installed, but not running try:
```

Comment:
All of the instructions here will need more careful analysis, as they are essentially documentation.